sklearn/datasets/base.py

```diff
@@ -764,3 +764,108 @@ def _pkl_filepath(*args, **kwargs):
         basename += py3_suffix
     new_args = args[:-1] + (basename + ext,)
     return join(*new_args)
+
+
+class partialURLOpener(urllib.FancyURLopener):
+    """
+    Override HTTP Error 206 (partial file being sent)
+    """
+    def http_error_206(self, url, fp, errcode, errmsg, headers, data=None):
+        # Ignore the expected "error" code
+        pass
+
+
+def md5(path):
+    """
+    Calculate the md5 hash of the file at path.
+
+    Parameters
+    -----------
+    path: String
+        Path of file to calculate MD5 hash of.
+
+    Returns
+    -------
+    md5hash : String
+        MD5 hash of the file at the provided path.
+
+    """
+
+    md5hash = hashlib.md5()
+    md5hash.update(open(path, 'rb').read())
```

Comment:
You might want to read and update in chunks if we're dealing with any big datasets.